packages/pangea-node-sdk/src/services/authz.ts

```diff
@@ -0,0 +1,172 @@
+import PangeaResponse from "@src/response.js";
+import BaseService from "./base.js";
+import PangeaConfig from "@src/config.js";
+import { AuthZ } from "@src/types.js";
+
+/**
+ * AuthZService class provides methods for interacting with the AuthZ Service
+ * @extends BaseService
+ */
+class AuthZService extends BaseService {
+  constructor(token: string, config: PangeaConfig) {
+    super("authz", token, config);
+  }
+
+  /**
+   * @summary Tuple Create
+   * @description Create tuples in the AuthZ Service. The request will fail if there is no schema
+   * or the tuples do not validate against the schema.
+   * @operationId authz_post_v1beta_tuple_create
+   * @param {AuthZ.TupleCreateRequest} request - An object representing the request to create tuples.
+   *    - {AuthZ.Tuple[]} request.tuples - List of tuples to be created.
+   * @returns {Promise} - A promise representing an async call to the tuple create endpoint.
+   * @example
+   * ```typescript
+   * const response = await authz.tuple_create({
+   *   tuples: [
+   *     {
+   *       resource: { namespace: 'folder', id: 'resource1' },
+   *       relation: 'editor',
+   *       subject: { namespace: 'user', id: 'user1' },
+   *     },
+   *     // Add more tuples as needed
+   *   ],
+   * });
+   * ```
+   */
+  tuple_create(
+    request: AuthZ.TupleCreateRequest
+  ): Promise<PangeaResponse<AuthZ.TupleCreateResult>> {
+    return this.post("v1beta/tuple/create", request);
+  }
+
+  /**
+   * @summary Tuple List
+   * @description List tuples in the AuthZ Service based on provided filters.
+   *
+   * @operationId authz_post_v1beta_tuple_list
+   * @param {AuthZ.TupleListRequest} request - An object representing the request to list tuples.
+   *    - {AuthZ.TupleListFilter} request.filter - Filter object to narrow down the list of tuples.
+   *    - {number} [request.size] - Maximum results to include in the response. Minimum is `1`.
+   *    - {string} [request.last] - Reflected value from a previous response to obtain the next page of results.
+   *    - {AuthZ.ItemOrder} [request.order] - Order results asc(ending) or desc(ending).
+   *    - {AuthZ.TupleOrderBy} [request.order_by] - Which field to order results by.
+   * @returns {Promise} - A promise representing an async call to the tuple list endpoint.
+   * @example
+   * ```typescript
+   * const response = await authz.tuple_list({
+   *   filter: {
+   *     resource: { namespace: 'folder', id: 'resource1' },
+   *   },
+   *   size: 10,
+   * });
+   * ```
+   */
+  tuple_list(request: AuthZ.TupleListRequest): Promise<PangeaResponse<AuthZ.TupleListResult>> {
```

Comment:
Currently the only service that has subclasses is AuthN due to it's the bigger one I think, so in order to avoid a lot of prefixes we created subclasses, but it's not used in any other service. Particularly in AuthZ that we have 6 endpoints I don't think we really need it

Reply:
I guess I was more just thinking if there should consistency whee there is a `/` in the endpoint results in `.` in the sdk. But if don't do this for other cases then this is fine, although similar to listSubjects should this be tupleList

Reply:
Yes, name was already updated to `tupleList`